source/simulator/assembly.cc

```diff
@@ -943,6 +943,19 @@ namespace aspect
                                                           scratch.finite_element_values,
                                                           introspection);
 
+    if (advection_field.is_temperature())
+      scratch.material_model_inputs.requested_properties
+        =
+          MaterialModel::MaterialProperties::equation_of_state_properties |
+          MaterialModel::MaterialProperties::additional_outputs |
+          MaterialModel::MaterialProperties::viscosity |
+          MaterialModel::MaterialProperties::thermal_conductivity;
+
+    for (const auto &heating_model : heating_model_manager.get_active_plugins())
+      scratch.material_model_inputs.requested_properties
+        = scratch.material_model_inputs.requested_properties |
+          heating_model->get_required_properties();
```

Comment:
This is good, but can you add it to the following two places as well:

- in line 1064 we do the same for the face terms (only relevant for DG fields)
- in entropy_viscosity.cc:564 we do another material model evaluation for the entropy viscosity computation, this would also benefit from the adjustment (and should also affect your runtime).

Reply:
> This is good

I looked at this again with fresh eyes today, and now I think we don't even need the viscosity and the additional outputs (the heating models can add them if they need them, but none of the assemblers themselves include them).

Reply:
This PR was just about the temperature equation. You are right, there is also the Stokes preconditioner/GMG, and then there are the postprocessors (since my reactions happen on the particles, I technically do not need to compute them there either). I thought I'd look into this in separate PRs.